### ISSUE TITLE
I can't see the letters in the armory 3D user interface

### ISSUE THREAD

**Issue by jesusteama** [armory3d/armory#2026]
When I press edit canvas in the user interface I do not get the letters, but some strange symbols, help me, how could I solve this.

![](https://user-images.githubusercontent.com/75128834/100459586-b3373e80-3093-11eb-8fde-d5b1b730da16.jpg)

**Comment by knowledgenude** [comment]
Everything is fine here, are you using the Git version of Armory?

**Comment by MoritzBrueckner** [comment]
I have the same issue on my laptop with an AMD Radeon graphics card, however on my main computer with an Nvidia card everything works fine. @jesusteama Could you please post your computer specs (graphics card etc.?)

For me it doesn't work on:
Windows 10
CPU: AMD Ryzen 7 Mobile 3700U
GPU: AMD Radeon RX Vega 10

@luboslenco could this be a Krom/Kha issue? I think there were similar problems in the past?

**Comment by knowledgenude** [comment]
@MoritzBrueckner , in RX 560 it is working. I guess it is happening just for Windows (or may be related to the drivers version, i don't know). Here is a commit that looks like changes something in Windows: https://github.com/armory3d/armory2d/commit/8f48e88e26d9e2d268259747e5b678afc232be66

**Comment by jesusteama** [comment]
computer specs:
CPU: AMD Rysen 5 3400G with Raedon Vega Graphics 3.70GHz
RAM: 8,00GB
the computer does not have a GPU
Windows 10 64bits

**Comment by jesusteama** [comment]
what I did was download the SDK from the armory page, unzipped the file, went to blender 2.83, but I have it as .zip, I don't know if that affects, then I went to edit preferences and installed the armory.py file and put a location in the SDK path. I don't know if I should do something else.

**Comment by MoritzBrueckner** [comment]
Your installation seems fine and unreated to this, it rather looks like a driver issue or something is wrong with the text rendering backend.

**Comment by luboslenco** [comment]
> @luboslenco could this be a Krom/Kha issue? I think there were similar problems in the past but I can't find the issue

@MoritzBrueckner Yes I recall the issue but never got it to reproduce iirc. Is the text in debug console working fine for you on that machine? Does increasing the font size fix it?

**Comment by MoritzBrueckner** [comment]
Its the same with the debug console, but increasing the font size indeed fixes the problem.

**Comment by jesusteama** [comment]
How can I do that? (increase font size)

**Comment by MoritzBrueckner** [comment]
@jesusteama It's something you have to do in the code I think. For the debug console, go to the following lines in `yourSDKFolder/armory/Sources/armory/trait/internal/DebugConsole.hx` and add a line `ui.fontSize = [someFontSizeThatFitsYourNeed];` directly after `function render2D(g: kha.graphics2.Graphics) {`.

However, for Armory2D you would have to change it on some different place (didn't look after that yet) and compile it yourself.
Maybe changing the UI scale in Armory's preferences does help as well, I didn't test it.

But please note that this is only a workaround, not a fix!

**Comment by luboslenco** [comment]
> Maybe changing the UI scale in Armory's preferences does help as well, I didn't test it.

Yes, that will do it. 👍

**Comment by MoritzBrueckner** [comment]
I'm reopening this issue as the bug is not fixed yet, we only have a workaround so far :)

**Comment by MoritzBrueckner** [comment]
I can still reproduce this issue (AMD Radeon RX 6700 XT), in some (rare) cases the text looks as if the glyph quads draw a part from the glyph texture that is between multiple glyphs:
![Screenshot](https://github.com/armory3d/armory/assets/17685000/04747df0-b9ca-4007-bdbe-179b6a526348)
In this picture (from the "ms" tab label in the debug console) you can see a "20" at the top left. I feel like this effect mostly occurs with numbers, regular text seems more "scrambled" and you can mostly (but not only) see vertical lines. Since the issue seems as if there is some problem with sampling the glyph texture at the correct coordinates, the vertical lines could be occurring because many characters somehow end up in the bottom part of the glyph texture which contains empty boxes:
![Glyph Texture](https://github.com/armory3d/armory/assets/17685000/a8972d32-1eea-4100-a044-7b45604fd95f)

Unfortunately I'm not able to reproduce the issue in Renderdoc, the text is rendered just fine there. The same holds true for html5 and windows-hl.

I mentioned in a post above that I could remember other issues for this problem, and I finally found them. Sadly, they didn't mention a proper solution, but I'm still linking to them for completeness:
- https://github.com/armory3d/armortools/issues/70
- https://github.com/armory3d/armory/issues/1397
